crates/router/src/connector/datatrans/transformers.rs

```diff
@@ -82,23 +93,58 @@ pub enum DataTransCancelResponse {
     Empty,
 }
 
-#[derive(Serialize, Deserialize, Debug)]
+#[derive(Serialize, Deserialize, Clone, Debug)]
 #[serde(rename_all = "camelCase")]
 pub struct SyncResponse {
     pub transaction_id: String,
     #[serde(rename = "type")]
     pub res_type: TransactionType,
     pub status: TransactionStatus,
+    pub detail: SyncDetails,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct SyncDetails {
+    fail: Option<FailDetails>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct FailDetails {
+    reason: Option<String>,
+    message: Option<String>,
 }
 
-#[derive(Serialize, Clone, Debug)]
+#[derive(Serialize, Debug)]
 #[serde(rename_all = "camelCase")]
 pub struct PlainCardDetails {
     #[serde(rename = "type")]
     pub res_type: String,
     pub number: cards::CardNumber,
     pub expiry_month: Secret<String>,
     pub expiry_year: Secret<String>,
+    pub cvv: Secret<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(rename = "3D")]
+    pub three_ds: Option<ThreedsInfo>,
+}
+
+#[derive(Serialize, Debug)]
+pub struct ThreedsInfo {
+    cardholder: CardHolder,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CardHolder {
+    cardholder_name: Secret<String>,
+    email: Email,
+    bill_addr_line1: Secret<String>,
+    bill_addr_post_code: Secret<String>,
+    bill_addr_city: String,
+    bill_addr_state: Secret<String>,
+    bill_addr_country: common_enums::CountryAlpha2,
```

Comment:
Make only required fields as mandatory and add the in `crates/router/src/configs/defaults.rs`.
Make strongly recommended fields as optional and add a comment highlighting those fields.

Reply:
Updated